core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
@stew Yeah I had wondered that we have `Kleisli#local: Kleisli[M, A, B] => (AA => A) => Kleisli[M, AA, B]` (which I find useful in "expanding" the config when I use it in a more global scope), but also `MonadReader#local: M[R, A] => (R => R) => M[R, A]`. Perhaps we can rename the `Kleisli#local` to something else? ... `expand` ?